utils/interface-validator.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2023 Canonical
+# See LICENSE file for licensing details.
+
+"""Schem validator for `interface.yaml`"""
+import yaml
+import glob
+import re 
+
+from enum import Enum
+from typing import List, Optional
+from pydantic import AnyHttpUrl, BaseModel, Field, ValidationError
+
+
+class StatusEnum(str, Enum):
+    draft = "draft"
+    live = "live"
+    retired = "retired"
+
+class CharmEntry(BaseModel):
```

Comment:
this is missing some optional fields that can be used to configure pytest-interface-tester. See https://github.com/canonical/pytest-interface-tester?tab=readme-ov-file#customizing-the-fixtures-address